test/integration_test.py

```diff
@@ -641,6 +630,56 @@ def worker(_):
                     expect_execution = fetch == first_fetch
                     self.assertEqual(1 if expect_execution else 0, len(execution_ids))
 
+    def _manifest_filters(self, catalog: CatalogName) -> JSON:
+        # IT catalogs with just one public source are always indexed
+        # completely if that source contains less than the minimum number of
+        # bundles required. So regardless of any randomness employed by this
+        # test, manifests derived from these catalogs will always be based
+        # on the same content hash. Since the resulting reuse of cached
+        # manifests interferes with this test, we need another means of
+        # randomizing the manifest key: a random but all-inclusive filter.
+        tibi_byte = 1024 ** 4
+        return {
+            self._file_size_facet(catalog): {
+                'within': [[0, tibi_byte + self.random.randint(0, tibi_byte)]]
+            }
+        }
+
+    def _test_manifest_tagging_race(self, catalog: CatalogName):
+        validators: dict[ManifestFormat, Callable[[str, bytes], None]] = {
+            ManifestFormat.compact: self._check_compact_manifest,
+            ManifestFormat.curl: self._check_curl_manifest,
+        }
+
+        for format, validator in validators:
```

Comment:
Did you test this? I think it needs to be:
```suggestion
        for format, validator in validators.items():
```